File: tyama_icpc2012dA.clj
#!/usr/bin/clojure

(defn run [n]
	(if (not (zero? n))
		(let [y (read) m (read) d (read)]
			(println (- 196471 (+ (* (- y 1) 195 ) (* (quot (- y 1) 3) 5) (* (- m 1) 20) (- (if (zero? (rem y 3)) 0 (quot (- m 1) 2))) d)))
			(run (- n 1))
		)
	)
)

(let [n (read)]
	(run n)
)

File: tyama_icpc2012dA.hs
#!/usr/bin/env runghc
import Control.Applicative

run n
		| n>0 = do
			[y,m,d] <- map read . words <$> getLine
			print $ 196471-((y-1)*195)-(div (y-1) 3)*5-(m-1)*20+(if ((mod y 3)==0) then 0 else (div (m-1) 2))-d
			run (n-1)
		| otherwise = return ()

main = readLn >>= run